src/main/java/nextstep/subway/line/domain/Sections.java

```diff
@@ -0,0 +1,68 @@
+package nextstep.subway.line.domain;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+import javax.persistence.Embeddable;
+import javax.persistence.OneToMany;
+
+import nextstep.subway.station.domain.Station;
+
+@Embeddable
+public class Sections {
+
+	@OneToMany(mappedBy = "line")
+	private List<Section> sections = new ArrayList<>();
+
+	public Sections() {
+	}
+
+	public Sections(List<Section> sections) {
+		this.sections = sections;
+	}
+
+	public void add(Section section) {
+		sections.add(section);
+	}
+
+	public List<Section> getSections() {
+		return sections;
+	}
+
+	public List<Station> getStations() {
+		Station startStation = findStartStation();
+		return getOrderedStations(startStation);
+	}
+
+	private List<Station> getOrderedStations(Station startStation) {
+		Map<Station, Station> upDownStations = getUpDownRoute();
+		List<Station> stations = new ArrayList<>();
+		Station nextStation = startStation;
+		while (upDownStations.containsKey(nextStation)) {
+			stations.add(nextStation);
+			stations.add(upDownStations.get(nextStation));
+			nextStation = upDownStations.get(nextStation);
+		}
+		return stations;
+	}
+
+	private Map<Station, Station> getUpDownRoute() {
+		return sections.stream()
+			.collect(Collectors.toMap(entry -> entry.getUpStation(), entry -> entry.getDownStation(),
+				(o1, o2) -> o1, HashMap::new));
+	}
+
+	private Station findStartStation() {
+		Set<Station> downStations = sections.stream().map(it -> it.getDownStation()).collect(Collectors.toSet());
```

Comment:
```suggestion
		Set<Station> downStations = sections.stream()
				.map(Section::getDownStation)
				.collect(Collectors.toSet());
```
여기도 메서드 레퍼런스 활용!
그리고 앞선 미션에서 확인했었던 객체 지향 생활 체조 원칙 규칙 4번을 만족시키면 좋을 것 같아요~
`규칙 4: 한 줄에 점을 하나만 찍는다.`

Reply:
메서드 레퍼런스와 객체지향 생활체조 원칙은 한번 짚어주신 부분인데 챙기지 못하였네요 🥲
한번 짚어주신 부분은 상기해서 더 꼼꼼히 하도록 하겠습니다.